### PR TITLE
ROC-2835 - add a timeout for to allow uploading of large files

### PR DESCRIPTION
**Pull request by robertparkinson**

### CODE REVIEW

File: src/main/app/forms/models/fileTypeErrors.ts [hunk 1]
```diff
@@ -16,7 +17,9 @@ export class FileTypeErrors {
   static all (): FileTypeErrors[] {
     return [
       FileTypeErrors.WRONG_FILE_TYPE,
-      FileTypeErrors.FILE_TOO_LARGE
+      FileTypeErrors.FILE_TOO_LARGE,
+      FileTypeErrors.FILE_REQUIRED,
+      FileTypeErrors.REQUEST_TIMEOUT
```
Comment (rlewan) [inline src/main/app/forms/models/fileTypeErrors.ts]: That's not a part of **file type** errors.

Comment (robertparkinson) [reply src/main/app/forms/models/fileTypeErrors.ts]: reworded for clarity

File: src/main/features/certificateOfService/routes/file-upload.ts [hunk 2]
```diff
@@ -15,12 +15,18 @@ export default express.Router()
   .post(Paths.fileUploadPage.uri,
     ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
       const form = new formidable.IncomingForm()
+      const user: User = res.locals.user
       const FILE_SIZE_LIMIT: number = 10485760
       form.keepExtensions = true
+      req.setTimeout(7200000,function () {
```
Comment (rlewan) [inline src/main/features/certificateOfService/routes/file-upload.ts]: Extract this magic number to a constant with a name that makes it clear what the timeout value is.

File: src/main/features/certificateOfService/routes/file-upload.ts [hunk 3]
```diff
@@ -15,12 +15,18 @@ export default express.Router()
...
+        user.legalUploadDocumentDraft.document.fileToUploadError = FileTypeErrors.FILE_UPLOAD_TIMEOUT
```
Comment (rlewan) [inline src/main/features/certificateOfService/routes/file-upload.ts]: Shouldn't the timeout error be used here?

**Pull request merged**